### PR TITLE
Preference loading tweaks

### PR DESCRIPTION
**Pull request by natewalck**
This does two major things:

- Add info logging for missing osVersionRequirements key
- Read prefs only once in DefaultPreferencesNudge

### CODE REVIEW

File: Nudge/Preferences/DefaultPreferencesNudge.swift [hunk 1]
```diff
@@ -7,17 +7,21 @@
 
 import Foundation
 
+// Try to grab settings from profile and json
+let osVersionRequirementsProfile = getOSVersionRequirementsProfile() ?? nil
```
Comment (natewalck) [inline Nudge/Preferences/DefaultPreferencesNudge.swift:11]: Might not need none here as the function returns none if there are no prefs

**Comment by erikng** [comment]
If you're going to do something like this, it needs to be consistent on _all_ keys for Nudge.

File: Nudge/Preferences/DefaultPreferencesNudge.swift (hunk 1 above)
Comment (erikng) [reply Nudge/Preferences/DefaultPreferencesNudge.swift:11]: yeah this should already return nil since we do `??` in all the keys. It's an optional value

**Comment by erikng** [comment]
@natewalck can you also move this to a PR on `development` branch? That branch has automated testing for new diffs.

**Comment by erikng** [comment]
Since you split this diff, you now need to rebase.

**Comment by erikng** [comment]
I don't understand why the CI is failing other than I think it's running on your repo and not the you you are PR'ing to.

That said the big problem I have here is it changes the behavior of how nudge operates. Currently you can deploy both a profile and json and manage different keys. I'm not sure how many people would be in this state, but it is different behavior than what I personally intended.

Am I reading this wrong or am I indeed correct?

**Comment by erikng** [comment]
According to https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/ with the latest change I did, this should trigger the right CI job. @natewalck can you rebase and try again?

**Comment by erikng** [comment]
Alright CI is working and has passed, now back to

> That said the big problem I have here is it changes the behavior of how nudge operates. Currently you can deploy both a profile and json and manage different keys. I'm not sure how many people would be in this state, but it is different behavior than what I personally intended.

**Comment by natewalck** [comment]
I verified that the settings blocks are properly read and work.

File: Nudge/Utilities/Preferences.swift [hunk 2]
```diff
@@ -91,14 +109,60 @@ func getAboutUpdateURL() -> String? {
     return ""
 }
 
+// userExperience
+// Even if profile is installed, return nil if in demo-mode
```
Comment (erikng) [inline Nudge/Utilities/Preferences.swift:113]: can you change this to `// Even if profile/JSON` since you do the same in the function below?

**Review by erikng — COMMENTED** [review]
One tiny change and we can merge

**Pull request merged**